risc0/zkvm/src/receipt.rs

```diff
@@ -173,45 +170,141 @@ pub struct SessionReceipt {
     pub journal: Vec<u8>,
 }
 
-/// Provide common functionality implemented by all receipt types.
-#[dyn_partial_eq]
-#[typetag::serde(tag = "type")]
-pub trait Receipt: Debug {
+/// An inner receipt can take the form of a collection of [SegmentReceipts] or a
+/// [SuccinctReceipt].
+#[derive(Debug, Deserialize, Serialize, PartialEq)]
+pub enum InnerReceipt {
```

Comment:
Since we won't have a `Receipt` trait, should we simply call this enum `Receipt`?

Reply:
If anything, I was thinking we'd want to rename `SessionReceipt` to just `Receipt` since that's basically what users are going to interact with.